src/src_sinc.c

```diff
@@ -31,9 +31,20 @@
 typedef int32_t increment_t ;
 typedef float	coeff_t ;
 
+#ifdef _MSC_VER
+#define MSVC_DISABLE_WARNING(n) __pragma(warning(push)) \
+								__pragma(warning(disable:n))
```

Comment:
Can't this be done in CMake somewhere?

Reply:
Yes, but that would be more work for no gain. MSVC guarantees that `_MSC_VER` is defined

Work: Add configured define to `config.h` if compiling with MSVC   
Gain: Can use `#if SRC_MSVC` instead of the `#ifdef _MSC_VER` where the former would error on compile time if it is not set catching e.g. a typo in `_MSC_VER`

But: If `_MSC_VER` was mistyped then the warnings won't be disabled and hence CI would fail due to the warnings thrown. Therefore a mistyped macro name would already result in a compile-time error which is what is the goal of using `#if` over `#ifdef`, but using `_MSC_VER` is less work

Reply:
I would *vastly* prefer to limit compiler specific specical case handling creeping in the code and if it does creep into the code it should be mininal. This approach does not meet those goals.

Reply:
Oh yes, I could completely disable this warning in CMake if this is what you meant. I had done that initially: https://github.com/erikd/libsamplerate/pull/82/commits/b4873245295b2bc5b9ba0d13ef4b7fd584efd94a#diff-af3b638bc2a3e6c650974192a53c7291R113
But earlier you said to use the pragma instead: https://github.com/erikd/libsamplerate/pull/82#discussion_r308597730

My preferred approach would be to use the proper explicit casts and `f` suffix for float constants. This makes the truncation obvious which might be important knowledge when changing the code. This would also never required warning suppressions on any compiler

Reply:
I added another commit which does just that: Explicit casts where truncation happens. This makes the compilers happy so no suppression required. https://github.com/erikd/libsamplerate/pull/82/commits/37473c28386c1c5dcc0352759ce6e78a93f030e5

Just choose which one you want:
1. warning suppression flag added in CMake
2. pragma based suppression
3. explicit handling of narrowing conversions (my preference)

Notes:
- It might be worth checking if some of those calculations should really be done using `double` instead of `float`.
- Loops are used instead of manual unrolling. The compiler is at least as good and in this case generates the same machine code with less code and room for error: https://godbolt.org/z/4a27qn
- To kinda prove the above: The manually unrolled loops in `calc_output_multi` are suboptimal due to incorrect tail handling: The `switch (ch % 8)` has to be place on top of the `do` otherwise the condition and conditional jump will be executed on every iteration instead of once. I'd suggest to just use a loop and let the compiler decide.